### PR TITLE
docs(breaking): add v8 browser and platform support

### PR DESCRIPTION
**Pull request by liamdebeasi**
Issue number: N/A

---------

## What is the current behavior?

As part of our Ionic 8 work, the team would like to define which browsers and platforms we will support.

## What is the new behavior?

- Chromium (Chrome + Edge) support has changed from 79+ to 89+
- Firefox support has changed from 70+ to 75+
- Safari/WebKit support has changed from 14+ to 15+
- Angular support has changed from 14+ to 16+

A few notes on why I made these changes:

- These browsers have full support for [Web Animations Level 1](https://caniuse.com/?search=web%20animations). This allows us to remove the old CSS Animation fallback support from our animation library.
- These browsers have [native image lazy loading](https://caniuse.com/?search=image%20lazy%20loading) support (iOS has it starting in iOS 15.4) which allows us to reduce our dependency on `ion-img`.
- These browsers have full support for [CSS Logical Properties](https://caniuse.com/?search=logical%20properties) which will help improve our RTL support in Ionic.
- This also gives us ~3 years of Firefox and Chrome support by the time Ionic 8 comes out. Firefox 75 came out April 2020, and Chrome 89 came out February 2021.
- Dropping iOS 14 allows us to remove several iOS 14-related hacks in item, datetime, and more.
- Dropping iOS 14 still lets us support >94% of iPhones and >91% of iPads as of May 2023 (https://developer.apple.com/support/app-store/).
- Dropping Angular 14 and 15 allows us to remove several Angular specific hacks. Angular 14 LTS ends at the end of 2023, and Angular 15 LTS ends in May 2024, so these versions are not being supported by the Angular team for much longer. By the time Ionic 8 is out, Angular 17 should be released with Angular 18 coming soon.
## Does this introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: BREAKING.md [hunk 1]
```diff
@@ -12,3 +12,30 @@ This is a comprehensive list of the breaking changes introduced in the major ver
...
+| Safari          | 15+               |
+| Firefox         | 75+               |
+| Edge            | 84+               |
+
+**Minimum JavaScript Framework Versions**
+| Framework | Supported Version     |
+| --------- | --------------------- |
+| Angular   | 16+                   |
+| React     | 17+                   |
+| Vue       | 3.0.6+                |
```
Comment (liamdebeasi) [inline BREAKING.md:35]: I didn't see a big need to change these versions, so I left them alone for now.

**Review by brandyscarney — APPROVED** [review]

**Review by thetaPC — APPROVED** [review]
LGTM

**Comment by liamdebeasi** [comment]
Re-requesting the team for review.

I wanted to propose changing the Chromium minimum version to 89+ instead of the previously noted 84+. The reason for this is targeting 89+ allows us to make use of [CSS Logical Properties](https://caniuse.com/?search=logical%20properties) for improved RTL support. We make use of some today, but we'd be able to use even more starting in v8. This could also reduce our dependence on custom mixins for certain properties.

**Review by brandyscarney — APPROVED** [review]

**Review by thetaPC — APPROVED** [review]
LGTM

**Pull request merged**